--- target/linux/ramips/image/mt7621.mk	2020-09-03 00:17:48.516763100 +0800
+++ target/linux/ramips/image/mt7621.mk	2020-09-02 23:37:16.570579100 +0800
@@ -215,7 +215,7 @@
 define Device/dlink_dir-8xx-a1
   IMAGE_SIZE := 16000k
   DEVICE_VENDOR := D-Link
-  DEVICE_PACKAGES := kmod-mt7615d luci-app-mtwifi uboot-envtools
+  DEVICE_PACKAGES := kmod-mt7615d-dual mt_wifi-mt7615-dual luci-app-mtwifi uboot-envtools
 endef
 
 define Device/dlink_dir-860l-b1
@@ -262,7 +262,7 @@
   DEVICE_VENDOR := Newifi
   DEVICE_MODEL := D2
   DEVICE_PACKAGES := kmod-mt7603e kmod-mt76x2e kmod-usb3 \
-	kmod-usb-ledtrig-usbport luci-app-mtwifi
+	kmod-usb-ledtrig-usbport mt_wifi-mt7603-mt7612 luci-app-mtwifi
 endef
 TARGET_DEVICES += d-team_newifi-d2
 
@@ -361,7 +361,7 @@
   DEVICE_VENDOR := GeHua
   DEVICE_MODEL := GHL-R-001
   DEVICE_PACKAGES := kmod-mt7603e kmod-mt76x2e kmod-usb3 \
-	kmod-usb-ledtrig-usbport luci-app-mtwifi
+	kmod-usb-ledtrig-usbport mt_wifi-mt7603-mt7612 luci-app-mtwifi
 endef
 TARGET_DEVICES += gehua_ghl-r-001
 
@@ -602,7 +602,7 @@
   IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Motorola
   DEVICE_MODEL := MR2600
-  DEVICE_PACKAGES := kmod-mt7615d kmod-usb3 kmod-usb-ledtrig-usbport luci-app-mtwifi uboot-envtools
+  DEVICE_PACKAGES := kmod-mt7615d-dual kmod-usb3 kmod-usb-ledtrig-usbport mt_wifi-mt7615-dual luci-app-mtwifi uboot-envtools
 endef
 TARGET_DEVICES += motorola_mr2600
 
@@ -730,7 +730,7 @@
   SERCOMM_HWVER := A001
   SERCOMM_SWVER := 0x0052
   IMAGE_SIZE := 40960k
-  DEVICE_PACKAGES += -kmod-mt7603 -wpad-openssl kmod-mt7603e kmod-mt7615d luci-app-mtwifi
+  DEVICE_PACKAGES += -kmod-mt7603 -wpad-openssl kmod-mt7603e kmod-mt7615d-single mt_wifi-mt7603-mt7615 luci-app-mtwifi
 endef
 TARGET_DEVICES += netgear_r6850
 
@@ -805,7 +805,7 @@
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := K2P
   SUPPORTED_DEVICES += k2p
-  DEVICE_PACKAGES := kmod-mt7615d luci-app-mtwifi
+  DEVICE_PACKAGES := kmod-mt7615d-dbdc mt_wifi-mt7615-dbdc luci-app-mtwifi
 endef
 TARGET_DEVICES += phicomm_k2p
 
@@ -987,7 +987,7 @@
   SUPPORTED_DEVICES += R3G
   SUPPORTED_DEVICES += mir3g
   DEVICE_PACKAGES := kmod-mt7603e kmod-mt76x2e kmod-usb3 \
-	kmod-usb-ledtrig-usbport luci-app-mtwifi uboot-envtools
+	kmod-usb-ledtrig-usbport  mt_wifi-mt7603-mt7612 luci-app-mtwifi uboot-envtools
 endef
 TARGET_DEVICES += xiaomi_mir3g
 
@@ -1037,7 +1037,7 @@
   DEVICE_MODEL := Mi Router 4
   SUPPORTED_DEVICES += R4
   SUPPORTED_DEVICES += mir4
-  DEVICE_PACKAGES := kmod-mt7603e kmod-mt76x2e luci-app-mtwifi uboot-envtools
+  DEVICE_PACKAGES := kmod-mt7603e kmod-mt76x2e mt_wifi-mt7603-mt7612 luci-app-mtwifi uboot-envtools
 endef
 TARGET_DEVICES += xiaomi_mir4
 
@@ -1054,7 +1054,7 @@
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | check-size
   DEVICE_VENDOR := Xiaomi
-  DEVICE_PACKAGES := kmod-mt7603e kmod-mt7615d luci-app-mtwifi uboot-envtools
+  DEVICE_PACKAGES := kmod-mt7603e kmod-mt7615d-single mt_wifi-mt7603-mt7615 luci-app-mtwifi uboot-envtools
 endef
 
 define Device/xiaomi_mi-router-ac2100
